sdk/servicebus/microsoft-azure-servicebus/src/main/java/com/microsoft/azure/servicebus/management/ManagementClientAsync.java

```diff
@@ -92,9 +92,48 @@ public ManagementClientAsync(URI namespaceEndpointURI, ClientSettings clientSett
         DefaultAsyncHttpClientConfig.Builder clientBuilder = Dsl.config()
                 .setConnectTimeout((int) CONNECTION_TIMEOUT.toMillis())
                 .setRequestTimeout((int) this.clientSettings.getOperationTimeout().toMillis());
+
+        if(shouldUseProxy(this.namespaceEndpointURI.getHost())){
+            InetSocketAddress address = (InetSocketAddress)this.proxies.get(0).address();
+            String proxyHostName=address.getHostName();
+            int proxyPort=address.getPort();
+
+            clientBuilder.setProxyServer(new ProxyServer.Builder(proxyHostName,proxyPort));
+        }
```

Comment:
I am wondering if it may be easier/better to set the properties in the client that enables the integration with the default mechanisms that Java networking uses for proxy configuration - documented [here](https://docs.oracle.com/javase/8/docs/technotes/guides/net/proxies.html).  

```suggestion
                .setUseProxySelector(true)
                .setUseProxyProperties(true)
                .setRequestTimeout((int) this.clientSettings.getOperationTimeout().toMillis());
```

Reply:
@giventocode Thanks for pointing to this resource to offer more background. I verified that this change would have the same impact: select the default proxy set by ProxySelector, this is also what I was trying to do in my changes but don't aware that there are already system level support built into Java. I also verified this from here: https://www.javadoc.io/static/com.ning/async-http-client/1.9.31/com/ning/http/client/AsyncHttpClientConfig.Builder.html. It also discusses "If useProxySelector is set to true but setProxyServer(ProxyServer) was used to explicitly set a proxy server, the latter is preferred.", so should we just have .setUseProxySelector(true)? Have also verified that setting only this property will achieve the same goal in my test.

Reply:
Thanks for the additional insights. I further reviewed the code and in a nutshell: if there's a proxy selector, the proxy uri is retrieved from it and then is set explicitly as the proxy server to use.  So a few things to consider:

  - One of the benefits of implementing a proxy selector is that it allows you to filter/decide what proxy to use for a given URL - by overriding ``` java.util.List<Proxy> select(URI uri)```. The current implementation seems like it'd block this ability as the proxy selector won't be used at all.
 
  - Java also provides a standard way to set the proxy explicitly, as per below. It seems like this also won't be possible.  
```
System.setProperty("http.proxyHost", "webcache.example.com");
System.setProperty("http.proxyPort", "8080");
```  

As per the suggestion, if we set the flags in the underlying config of the asynchttpclient so it enables the default Java proxy setting behavior, then users can use the Java standard way of setting the proxies and extending how proxies are used programmatically.

Reply:
I have discussed with @giventocode offline and we have decided to add both `.setUseProxySelector(true)` (compatible with AMQP websockets to use ProxySelector for send/receive operations) and `.setUseProxyProperties(true)` (users can use the Java standard way of setting the proxies and extending how proxies are used programmatically for ManagementClient.)